app/models/comment.rb

```diff
@@ -0,0 +1,17 @@
+class Comment < ApplicationRecord
+  belongs_to :user, optional: true
+  belongs_to :commentable, polymorphic: true
+  belongs_to :parent, optional: true, class_name: "Comment"
+
+  def comments
+    Comment.where(commentable: commentable, parent_id: id)
+  end
+
+  def destroy
+    update(user: nil, body: nil)
+  end
+
+  def deleted?
+    user.nil?
+  end
```

Comment:
What's the purpose of doing this?

Reply:
Delete comment

Reply:
Why do we have to modify destroy method for deleting comment?